### ISSUE TITLE
The `memray stats` output is misleading

### ISSUE THREAD

**Issue by godlygeek** [bloomberg/memray#135]
## Bug Report

**Current Behavior** 

The `stats` reporter shows a histogram labelled as:
```text
📏 Total allocations:
        2468520

📦 Total memory allocated:
        257.024MB

  Histogram of allocation size:
        min: 1.000B
        ---------------------------------------------
        < 4.000B   :    122 ▇
        < 24.000B  :  57425 ▇▇▇▇▇
        < 119.000B : 302549 ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
        < 588.000B :  93915 ▇▇▇▇▇▇▇▇
        < 2.829KB  :  26592 ▇▇▇
        < 13.933KB :   7789 ▇
        < 68.616KB :   1734 ▇
        < 337.926KB:    222 ▇
        < 1.625MB  :     13 ▇
        <=8.004MB  :     14 ▇
        ---------------------------------------------
        max: 8.004MB
```

But the histogram isn't _really_ "of allocation size", because the `stats` reporter is only seeing allocations after they've already been aggregated by the `SnapshotAllocationAggregator`. The `stats` reporter has absolutely no idea how big or small any single allocation was, because it only sees the sum of the sizes of the allocations performed by any given source location.

If we wanted this histogram to be correct, we'd need more information to survive the aggregation process: namely, the size of each individual allocation that contributed to the aggregated size associated with a source location.

Moreover, it's at least somewhat misleading that this histogram shows (by default) only allocations that haven't been freed. Nothing in the displayed stats indicates that the histogram is showing only allocated-but-not-freed things as of the high water mark, and in fact the labels above it ("Total allocations" and "Total memory allocated") make it sound like it is considering _all_ allocations, not just unfreed allocations as of the high water mark (I'd expect "Maximum memory allocated" rather than "Total memory allocated" to describe the latter case).

**Comment by godlygeek** [comment]
I suggest that we redo the `stats` reporter a bit. I think we should drop the `--include-all-allocations` option, and instead always process all allocations. Then the "Total allocations" and "Total memory allocated" labels would be correct. We could include as well "Maximum memory allocated" for free, since we learn that in our first pass when discovering the high water mark.

And I'd capture the actual sizes of the individual allocations (either with a 3rd pass, or by updating the `SnapshotAllocationAggregator` to record it), so that we can get a correct histogram, rather than binning the already-aggregated sizes.

**Comment by godlygeek** [comment]
This was closed by #136
